utils/urlencode.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015 Spinpunch, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"net/url"
+	"strings"
+)
+
+func UrlEncode(str string) string {
```

Comment:
Also the url.QueryEscape function works perfectly for this except it encodes spaces to "+" which gets all confused in the URL when there are any real "+" in the filename